privacy_analytics/middleware.py

```diff
@@ -1,26 +1,21 @@
-from .models import PageView
-import threading
-import queue
 from django.conf import settings
 
+from .models import PageView
+
+
 class AnalyticsMiddleware:
     def __init__(self, get_response):
         self.get_response = get_response
-        self.processing_queue = queue.Queue()
-        self.processing_thread = threading.Thread(target=self._thread_run)
-        self.processing_thread.daemon = True
-        self.processing_thread.start()
-
-    def _thread_run(self):
-        while True:
-           PageView.create_for_request(self.processing_queue.get())
-           # Will lock until item is available
 
     def __call__(self, request):
         response = self.get_response(request)
-        if hasattr(settings, "ANALYTICS_IGNORE_PATHS"):
-            for path in settings.ANALYTICS_IGNORE_PATHS:
-                if request.path.startswith(path):
-                    return response
-        self.processing_queue.put_nowait(request)
```

Comment:
Thank you for removing my very janky background queue system. That said, I'm a bit concerned about doing a db write on every page load. Seems like the sort of thing that could overload the server quite easily, as well as be vulnerable to a denial of service attack.

Would you consider moving the write system into a [Celery background task](http://www.celeryproject.org/)?

Reply:
Yes! I am going to do so, that is definitely next on the to-do list.

Reply:
I have added initial celery support; however, I would like to implement optional support for different task managers, but I am unsure on how to handle that as I've never done something like that before. Could you look into that?